src/entity/profession/ArmorerProfession.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\entity\profession;
+
+use pocketmine\block\VanillaBlocks;
+use pocketmine\data\bedrock\VillagerProfessionTypeIds;
+use pocketmine\entity\trade\TradeRecipe;
+use pocketmine\item\VanillaItems;
+use function random_int;
+
+final class ArmorerProfession extends VillagerProfession{
+
+	public function __construct(){
+		parent::__construct(VillagerProfessionTypeIds::ARMORER, "entity.villager.armor", VanillaBlocks::BLAST_FURNACE());
+	}
+
+	/** @phpstan-return list<TradeRecipe> */
+	public function getRecipes(int $biomeId) : array{
```

Comment:
In vanilla only a few recipes are selected per group, see https://github.com/Mojang/bedrock-samples/blob/main/behavior_pack/trading/armorer_trades.json, for more details. Probably with a function that generates recipes is not enough and will have to be extended to a dedicated class.

Reply:
So as far as I understand, (also according to [bedrock.dev wiki](https://wiki.bedrock.dev/loot/trade-tables.html#trade-groups)), recipes are separated something called "groups" and they are randomly selected whenever trade recipes are generated, right? Perhaps we could make a script that generates recipe class from behaviour pack.

Reply:
Yes, it would be best if this is self-generated. I think that for now we only need to make an api to implement trade tables with their groups and leave the generation of recipes for another PR.